### PR TITLE
Handle mixed-case columns in Elasticsearch

### PR DESCRIPTION
**Pull request by martint**

### CODE REVIEW

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java [hunk 1]
```diff
@@ -197,7 +199,9 @@ else if (type instanceof ObjectType) {
 
         ConnectorTableMetadata tableMetadata = getTableMetadata(session, tableHandle);
         for (ColumnMetadata column : tableMetadata.getColumns()) {
-            results.put(column.getName(), new ElasticsearchColumnHandle(column.getName(), column.getType()));
+            results.put(column.getName(), new ElasticsearchColumnHandle(
+                    column.getProperties().getOrDefault(ORIGINAL_NAME, column.getName()).toString(),
```
Comment (findepi) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java]: There is no reason to accept anything other than `String` here.

```suggestion
                    (String) column.getProperties().getOrDefault(ORIGINAL_NAME, column.getName()),
```

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java [hunk 2]
```diff
@@ -258,4 +262,15 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
 
         return Optional.of(new ConstraintApplicationResult<>(handle, constraint.getSummary()));
     }
+
+    private static ColumnMetadata makeColumnMetadata(String name, Type type)
+    {
+        return new ColumnMetadata(
+                name,
+                type,
+                "",
```
Comment (findepi) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java]:
```suggestion
                null,
```

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java [hunk 3]
```diff
@@ -258,4 +262,15 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
...
+                "",
```
Comment (findepi) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java]:
```suggestion
                null,
```

File: presto-elasticsearch/src/test/java/io/prestosql/elasticsearch/TestElasticsearchIntegrationSmokeTest.java [hunk 4]
```diff
@@ -304,6 +304,26 @@ public void testQueryString()
...
+
+        embeddedElasticsearchNode.getClient()
+                .admin()
+                .indices()
+                .refresh(refreshRequest(indexName))
+                .actionGet();
+
+        assertQuery(
+                "SELECT name, age FROM mixed_case",
+                "VALUES ('john', 32)");
```
Comment (findepi) [inline presto-elasticsearch/src/test/java/io/prestosql/elasticsearch/TestElasticsearchIntegrationSmokeTest.java]: Add a case with predicate on `name` being pushed down.

**Comment by martint** [comment]
@findepi, updated.

**Review by dain — APPROVED** [review]

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java [hunk 5]
```diff
@@ -258,4 +262,15 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
...
+
+    private static ColumnMetadata makeColumnMetadata(String name, Type type)
+    {
+        return new ColumnMetadata(
+                name,
+                type,
+                null,
+                null,
+                false,
+                ImmutableMap.of(ORIGINAL_NAME, name));
```
Comment (findepi) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java]: this should be

```
ColumnMetadata.builder()
  .setName(name)
  .setType(type)
  .setProperties(ImmutableMap.of(ORIGINAL_NAME, name))
  .builder();
```

**Review by findepi — APPROVED** [review]

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java (hunk 5 above)
Comment (martint) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java]: Oops. I missed this comment before I merged it. I'll update that and put up a separate PR.

**Pull request merged**